src/client/TerritoryPatterns.ts

```diff
@@ -35,28 +30,38 @@ class PatternDecoder {
     if (version !== 1) {
       throw new Error("The pattern versions are different.");
     }
-    const tileWidth = (bytes[1] << 8) | bytes[2];
-    const tileHeight = (bytes[3] << 8) | bytes[4];
-    const scale = (bytes[5] << 8) | bytes[6];
 
-    const totalBits = tileWidth * tileHeight;
-    const totalBytes = Math.ceil(totalBits / 8);
-    const data = bytes.slice(7, 7 + totalBytes);
-    console.log("data", data);
+    this.tileWidth = (bytes[1] << 8) | bytes[2];
+    this.tileHeight = (bytes[3] << 8) | bytes[4];
+    this.scale = (bytes[5] << 8) | bytes[6];
```

Comment:
The `Uint16array` class can operate as data view to directly read as a u16 without conversion:
```suggestion
    [this.tileWidth, this.tileHeight, this.scale] = new Uint16array(bytes, 1, 6);
```

Reply:
Note that x86 uses little endian byte order, which is reversed from what you have.